### DISCUSSION TITLE
Various issues with Helm chart

### DISCUSSION THREAD

**Discussion by gboor** [requarks/wiki#3686] [Error / Bug Report]
**Describe the bug**

I downloaded en used the most recent master version of the helm chart and ran into the following issues;

1) When installing as-is, helm complains that there is a port definition missing

```
Error: unable to build kubernetes objects from release manifest: error validating "": error validating data: ValidationError(Service.spec.ports[1]): missing required field "port" in io.k8s.api.core.v1.ServicePort
```
This is true, as line 19 in the service.yaml file reads;
`    - port: {{.Values.service.httpsPort}}`

but no default value for service.httpsPort is provided in the standard values.yaml; it is commented out. This can be bypassed by specifically setting service.httpsPort in your own values, but the whole port should probably be skipped if there is no definition for it.

2) image.tag does nothing; in the deployment.yaml on line 28 it reads `          image: "{{ .Values.image.repository }}:{{ .Chart.AppVersion }}"`, using .Chart.AppVersion for the tag instead of image.tag as defined in the values. The value of "latest" is very problematic for updating. I tried to set it to 2.4, but couldn't. The default should also be set to a fixed tag, not latest.

3) Deploying with an external postgresql does not work, because no secret is created with the password. In _helpers.tpl starting on line 88, the following block appears;

```
{{/*
Set postgres secret
*/}}
{{- define "wiki.postgresql.secret" -}}
{{- if .Values.postgresql.enabled -}}
{{- template "wiki.postgresql.fullname" . -}}
{{- else -}}
{{- template "wiki.fullname" . -}}
{{- end -}}
{{- end -}}
```

This sets the value of wiki.postgresql.secret to the wiki.fullname template (in my case "wikijs") if the coupled postgres is not installed. The secret, however, is not created anywhere and the container creation fails as a result.

A workaround for this is creating the secret by hand.

**To Reproduce**
1. Download the helm chart from the current master
2. Try to install it as-is
3. Try to install it with a custom tag
4. Try to install it with an external postgresql server

**Expected behavior**
Installation to work as documented

**Host Info (please complete the following information):**
 - OS: Ubuntu
 - Wiki.js version: 2.4
 - Database engine: Postgres 11

**Comment by jforman** [comment]
Image tag/version can be fixed by applying https://github.com/Requarks/wiki/pull/2472

**Comment by jochen-st** [comment]
How to create the secret by hand:

Use kubectl apply with the following:

~~~
apiVersion: v1
kind: Secret
metadata:
  name: wikijs-postgresql
data:
  postgresql-password: <your base64 encoded password goes here>
type: Opaque
~~~

and in the Helm chart set value `postgresql.existingSecret` to the secret name (`wikijs-postgresql` in this example)
